subsys/bluetooth/host/gatt.c

```diff
@@ -1474,7 +1474,7 @@ void bt_gatt_foreach_attr_type(uint16_t start_handle, uint16_t end_handle,
 			}
 
 			for (i = 0; i < static_svc->attr_count; i++, handle++) {
-				struct bt_gatt_attr attr;
+				static struct bt_gatt_attr attr;
```

Comment:
I don't understand why this fixes the issue (I'm assuming that you've tested with valgrind to validate this fix). Can you elaborate? 

I also wonder that if this fixes the issue here, do we need a similar addition of `static` to `foreach_attr_type_dyndb`?